press/press/doctype/server/server.py

```diff
@@ -811,9 +815,20 @@ def auto_scale_workers(self):
 			self._auto_scale_workers_old()
 
 	def _auto_scale_workers_new(self):
-		usable_ram = max(
-			self.ram - 3000, self.ram * 0.75
-		)  # in MB (leaving some for disk cache + others)
+		if self.ram < 8000:
+			usable_ram = self.ram - 2000  # in MB (leaving some for disk cache + others)
+		elif self.ram < 4000:
+			usable_ram = self.ram - 1500  # in MB (leaving some for disk cache + others)
+		else:
+			usable_ram = max(
+				self.ram - 3000, self.ram * 0.75
+			)  # in MB (leaving some for disk cache + others)
+		if self.is_self_hosted and (
+			self.name == self.database_server
+		):  # if app and database server are on same machine
+			usable_ram = (
+				self.ram - round(self.ram * 0.685)
+			) * 0.75  # [Current RAM - InnoDB Bufferpool Size] More ram for DB, use 75% of remaining ram for app
```

Comment:
Bufferpool shouldn't have 69% of all RAM when it's on app server, that balance is quite off. Maybe you'll have to fix mariadb config in case of single server db+app. 


Bufferpool is supposed to hold "working set of database" in memory. So if you have 1GB of database out of which 50% is frequently read/written then 500MB is good enough size for bufferpool. If you assume 69% and only 10% of it is actually in use then you'll end up wasting memory.

On BYOD where most common case is a single site running on server it doesn't need this much memory. The ratio 69% makes sense for a dedicated DB server because there it's JUST used for DB and nothing else, so whatever is theoretical max can be used.

---

IMO safe enough numbers to begin with:
- Compute usable ram first by subtracting with same old logic to preserve some headroom for memory-intensive tasks and other processes.
- 30-50% of usable RAM for DB
- 30-50% of usable RAM for app server


Also, some reconfigurability of these things will help a lot. (not server scripts :woozy_face: ) but overrides. 

Overrides are required because this math will mostly work fine but when it won't we will have to manually tweak it and make the tweaked numbers stick (right now they go away after bench update, which is very pointless configuration)

Reply:
I'm moving Worker Allocation Logic updates to Newer PR